### DISCUSSION TITLE
Welcome to the discussion forum!

### DISCUSSION THREAD

**Discussion by marcua** [b12io/orchestra#749] [General]
Hello everyone! We're trying a new experiment to centralize Orchestra discussion on this forum. Historically, we've split discussions across GitHub issues, Gitter, and a mailing list. Due to lack of centralization and certain mediums not fully matching our asynchronous followups, these efforts have largely not panned out. As part of migrating to a forum, here's the plan:
* a few of the core Orchestra developers are committed to replying to forum posts in a timely fashion.
* we'll use the forum as a space to make more detailed announcements and provide roadmap transparency. As an example, see [yesterday's post on autostaffing and a bunch of other goodies in v1.0.33](https://github.com/b12io/orchestra/discussions/748).
* we'll update our docs and website to make less/no mention of Gitter / the mailing list and message those channels that we're migrating to this forum.

Thank you for your interest in Orchestra, and in the spirit of discussions, send feedback and questions any time!

**Comment by shahroon** [comment]
Hi,

I am just testing the orchestra, I am able to setup the orchestra project successfully. I can see the admin area and users interface as well.

However, I am still unable to understand, how a user can add a task and assign to himself? Or to someone else? I can't see the option inside user interface just empty available tasks.

I am using the journalist workflow example, loaded all the data, have multiple types of users, but don't know how to "request a task" as suggested in tutorial.

Moreover, I was also trying to run the other instance of project simultaneously, as suggested in docs to run the journalist workflow with pre-defined steps. But getting bad signature error.

I appreciate the help.

Thanks

**Reply by gandhishripala** [reply]
Hi @shahroon ,

I am also setting up the journalist workflow, and have encountered the "Bad Signature" Error. I believe it was solved by setting up the absolute path in "GOOGLE_P12_PATH" in file orchestra_settings.py. Also, make sure to escape the directory separator, since I am using windows, I had to put in "D:\\<folder_name>\\<folder_name>\\<folder_name>\\<file_name>.p12".

Also, in file orchestra_settings.py, for "ORCHESTRA_PROJECT_API_SECRET", I had set value as "CHANGEME". Docs do mention to change it, but changing it to some random text is also giving the "Bad Signature" Error.

Hope this helps.

**Reply by shahroon** [reply]
Hi @gandhishripala,

Thanks for the help. However, it seems that wasn't the issue. Could you please share what you placed inside your api values? As per documentation, it seems like it was just a place holder CHANGEME  to be replaced by some "appropriate values" but not sure whether those are, just random values or something else.

Any help would be appreciated.

Thanks

**Reply by gandhishripala** [reply]
Hi @shahroon ,

Yes, docs to mention it to use "appropriate values", but nowhere I could find from where to get "appropriate values". Attaching screenshot of the API Values that I have used to run this project.

![image](https://user-images.githubusercontent.com/7397361/118604416-c3beab00-b7d2-11eb-9a8e-5116b00c195f.png)

Hope this helps.

Thank you.

**Reply by shahroon** [reply]
So now your both projects are running simultaneously with these values, without issues and the prefilled values from journalism flow are visible, when you login as editor and other users etc?

**Reply by gandhishripala** [reply]
both projects are running simultaneously - Yes
prefilled values from journalism flow are visible - No luck. When I load the data, it says that the tasks are loaded, but I need to manually re-run task assignment to receive task into existing account (I am recalling the error I read. Its not exact error). I tried to search to manually re-run task assignment, but got nothing.

Also, for interactive_simple_workflow_demo, it creates project successfully, demo gets started, it asks to login with user demo and complete the task, but I am not able to see the task.

I would say, for me, setup was successful but there were issues in the tasks loading.

**Reply by shahroon** [reply]
That's exactly happening with me too, except for me running the other project wasn't successful.

I also noticed in one of the commit around 25 days ago, they removed add task functionality from user interface. Perhaps one can only create through Admin interface and then we can move further from there.

In my opinion, the idea is good, however a complete comprehensive documentation is something that must be there to make it a success.

Thanks for the input, keep in touch if you find something relevant.

**Reply by adbharadwaj** [reply]
> @gandhishripala Re:  When I load the data, it says that the tasks are loaded, but I need to manually re-run task assignment to receive task into existing account (I am recalling the error I read. Its not exact error). I tried to search to manually re-run task assignment, but got nothing.

Can you provide some screenshots and more context about the error? I can probably help if you give some context or set of steps you followed prior to seeing the error!

**Reply by gandhishripala** [reply]
Sure @adbharadwaj . Thank you for helping me out. Sorry for delayed response. Here are the steps that I followed:

1. conda create -n orchestra35 python=3.5 (Created Virtual Env with Python Version 3.5)
2. conda activate orchestra35
3. cd Downloads/orchestra-main/ (Downloaded Orchestra Main Branch code and unzip it)
4. pip install -r requirements.txt
![Screenshot-1](https://user-images.githubusercontent.com/7397361/120888854-35db2080-c618-11eb-8da7-0761091a51d4.PNG)
5. django-admin startproject test_project
6. pip install orchestra
![Screenshot-2](https://user-images.githubusercontent.com/7397361/120888881-4e4b3b00-c618-11eb-8aa4-882f962885d5.PNG)
7. wget https://raw.githubusercontent.com/b12io/orchestra/stable/example_project/example_project/orchestra_settings.py
8. mv orchestra_settings.py test_project/test_project
9. nano test_project/test_project/orchestra_settings.py
![Screenshot-3](https://user-images.githubusercontent.com/7397361/120888900-6327ce80-c618-11eb-8ef1-129c74ead801.PNG)
![Screenshot-4](https://user-images.githubusercontent.com/7397361/120888901-6458fb80-c618-11eb-99b6-a4a2182942f1.PNG)
10. nano test_project/test_project/settings.py
11. Added Lines:
`
from .orchestra_settings import setup_orchestra
`
`
setup_orchestra(__name__)
`
12. wget https://raw.githubusercontent.com/b12io/orchestra/stable/example_project/example_project/urls.py
13. mv urls.py test_project/test_project
14. cd test_project
15. python manage.py migrate
![Screenshot-5](https://user-images.githubusercontent.com/7397361/120888941-8ce0f580-c618-11eb-8571-94b15bb8dd48.PNG)
16. python manage.py loadallworkflows
![Screenshot-6](https://user-images.githubusercontent.com/7397361/120888949-9702f400-c618-11eb-97f3-1cab3b83a3fb.PNG)
17. python manage.py loadworkflowsampledata simple_workflow/v1
18. python manage.py loadworkflowsampledata journalism/v1
19. python manage.py loaddata demo_admin
20. python manage.py runserver (Login with demo user was successful)
![Screenshot-7](https://user-images.githubusercontent.com/7397361/120888982-b732b300-c618-11eb-81a5-57ffb87bd18d.PNG)
21. python manage.py interactive_simple_workflow_demo (In another terminal, it created project with ID : 1)
![Screenshot-8](https://user-images.githubusercontent.com/7397361/120889025-e9dcab80-c618-11eb-8220-1f5b434ff61f.PNG)
22. Here, it asks to navigate to URL, login with demo User and click New Delivery Task Button
![Screenshot-9](https://user-images.githubusercontent.com/7397361/120889044-00830280-c619-11eb-8278-bd5e84cbb30f.PNG)
23. Unable to find the Button / Tasks
![Screenshot-10](https://user-images.githubusercontent.com/7397361/120889130-5bb4f500-c619-11eb-8f22-2a864bc305cc.PNG)
![Screenshot-11](https://user-images.githubusercontent.com/7397361/120889139-61123f80-c619-11eb-99c6-e9389c656b29.PNG)
![Screenshot-12](https://user-images.githubusercontent.com/7397361/120889147-68d1e400-c619-11eb-9543-575cbc0fbcf5.PNG)

Note:
1. During pip install requirements, getting issues with beanstalk-dispatch version conflicts with boto, Django, django-jsonview
2. Docs do mention changing settings.ORCHESTRA_PROJECT_API_SECRET = 'CHANGEME' to some appropriate value, but not sure where to get that "appropriate" value from. I tried putting in 'CHANGEME123', then it gives "Bad Signature" Error.
3. While loading all workflows, it says to manually re-run task creation logic.

I have tried to put in all the steps. OS is Ubuntu 20.04 and Python Version 3.5.

Let me know if any further information is required.

**Comment by DipaliJagtap07** [comment]
Hello everyone,

I hope this message finds you well. I wanted to bring to your attention an issue I've been experiencing while using the task management feature in our GitHub project. Specifically, I am encountering difficulties with viewing projects when adding tasks, as well as encountering a similar problem when attempting to add users from the user list.

When trying to add tasks, I'm unable to see the list of projects available for selection. Similarly, when adding users to tasks, the user list appears blank, making it impossible to select the desired users.

I've attached screenshots for reference [attach screenshots here]. As these issues significantly hinder our workflow and collaboration efforts, I wanted to reach out to see if others in the community have encountered similar challenges or if there are any known solutions or workarounds.

Your assistance in resolving this matter would be greatly appreciated. If there are any additional details or information needed to troubleshoot this issue, please don't hesitate to let me know.

Thank you in advance for your support.
![2024-05-22T14_06_59](https://github.com/b12io/orchestra/assets/118957488/84963932-78cc-4dc0-a545-bee3e6ddf5d3)

**Reply by marcua** [reply]
Hello @DipaliJagtap07! Thank you for trying Orchestra!

I see from your screenshot that you're using the Django Admin to try to create projects and tasks. Orchestra isn't quite so free-form---workflows and their steps are defined programmatically, and then programmatically turned into projects and tasks.

The [example workflow documentation](https://orchestra.readthedocs.io/en/latest/example_use.html) shows how to both define workflows/steps and create project/task instances from those definitions. And if you're searching for what it looks like to use the API, the [code underlying the Journalism example](https://github.com/b12io/orchestra/blob/198449272a96e17a9a55e1d002ff052b25a9d73e/journalism_workflow/management/commands/journalism_workflow_ctl.py#L51) shows how to create a project (which will create tasks) from a workflow.

I hope this helps. Thank you again!
